src/conspiracies/prompt_relation_extraction/template_class.py

```diff
@@ -0,0 +1,303 @@
+from abc import abstractmethod
+from typing import List, Tuple, Optional, Dict
+from spacy.tokens import Doc
+from conspiracies.prompt_relation_extraction.data_classes import SpanTriplet
+
+
+class Template:
+    """Abstract class for template classes. The class is used to create a
+    template for the triplet extraction task. The template is used to create a
+    prompt for the GPT-3 model.
+
+    Methods:
+        generate_prompt(target_tweet, generate_string):
+            creates a prompt for the GPT-3 model with the given target tweet. boolean option to generate the string over
+        set_examples(examples):
+            sets the examples to be used in the prompt
+        set_task_description(task_description):
+            sets the task description to be used in the prompt
```

Comment:
I haven't done this anywhere else in this repo, so I would probably move it to the function itself (but some people do add this here as well).